### ISSUE TITLE
Screen sharing with firefox on mac os

### ISSUE THREAD

**Issue by castfab** [muaz-khan/RTCMultiConnection#9]
Hello,

The screen sharing feature doesn't seem to work on firefox / mac os.

Tested with Firefox Nightly 35 on Mac OS Mavericks (screen sharing and allowed domains were set in about:config). Works well with nightly on Windows.

**Comment by u11140** [comment]
hello guys, got same error while working on firefox 35 please help me 
thank you.

**Comment by muaz-khan** [comment]
@u11140 did you modify and install Firefox extension?

* https://github.com/muaz-khan/Firefox-Extensions/tree/master/enable-screen-capturing

**Comment by u11140** [comment]
No. Actually I am using code of  "Screen Sharing" from https://www.webrtc-experiment.com/
I want to run this on my localhost. For that I have created a SSL Certificate on Apache for Ubuntu 14.04. After this what should I do ? When I run this program on localhost it gives me following error on console  " name PermissionDeniedError " "constraintName

{
	"audio": false,
	"video": {
		"maxWidth": 1920,
		"maxHeight": 1080,
		"mozMediaSource": "window",
		"mediaSource": "window"
	}
} "

"message Make sure that you are using Firefox Nightly and you enabled: media.getusermedia.screensharing.enabled flag from about:config page. You also need to add your domain in "media.getusermedia.screensharing.allowed_domains" flag. If you are using WinXP then also enable "media.getusermedia.screensharing.allow_on_old_platforms" flag. NEVER forget to use "only" HTTPs for screen capturing!" something like this.. 
So can you give me any suggestion for this?

**Comment by muaz-khan** [comment]
Modify this line and add `localhost` there:

* https://github.com/muaz-khan/Firefox-Extensions/blob/master/enable-screen-capturing/bootstrap.js#L18

Now, remake the XPI file. Drag/Drop XPI fie over Firefox address-bar, and install the extension.

Now, your screen capturing code will work in `https://localhost:port/`.

**Comment by u11140** [comment]
Still getting same error.. I changed domain to my domain name and also added "enable-screen-capturing.xpi"

**Comment by u11140** [comment]
Hello Sir,
I have done that. It is working fine now. But how should I run this xpi file from my website. I have tried Install Triggers function but it is not working . when click on button It gives me following error "the addon from localhost could not be installed because it does not match the add-on firefox expected  "

**Comment by muaz-khan** [comment]
* [bootstrap.js#L18](https://github.com/muaz-khan/Firefox-Extensions/blob/master/enable-screen-capturing/bootstrap.js#L18)

```javascript
['localhost'].forEach(function(domain) {
    if (values.indexOf(domain) === -1) {
        values.push(domain);
        addon_domains.push(domain);
    }
});
```

Now make ZIP file, rename "zip" to "xpi" and drag/drop "xpi" file over Firefox tab/address-bar.

**Comment by u11140** [comment]
On that way, it works fine but I want to give link from my website and when I click on that link it should install xpi file. I have given following link .
https://localhost:8443/xpi/firefox.xpi 
but it always give me "the add-on could not be downloaded because of connection failure on localhost"
